perf_benchmark/benchmark_main.cc

```diff
@@ -0,0 +1,435 @@
+                                                         chunk_per_msg);
+  }
+
+  // Benchmark the transcoding process
+  std::string message;
+  for (auto s: state) {
+    JsonRequestTranslator translator
+        (type_helper.Resolver(), is.get(), request_info, streaming, false);
+    MessageStream& out = translator.Output();
+
+    if (!out.Status().ok()) {
+      return absl::InternalError(out.Status().ToString());
+    }
+
+    while (out.NextMessage(&message)) {
+    }
+    is->Reset(); // low overhead.
+  }
+
+  // Add custom benchmark counters
+  auto request_processed = static_cast<double>(state.iterations());
+  auto message_processed =
+      static_cast<double>(state.iterations() * (streaming ? stream_size : 1));
+  auto bytes_processed =
+      static_cast<double>(state.iterations() * is->TotalBytes());
+  state.counters["byte_throughput"] = Counter(bytes_processed, Counter::kIsRate,
+                                              Counter::kIs1024);
+  state.counters["byte_latency"] =
+      Counter(bytes_processed,
+              Counter::kIsRate | Counter::kInvert,
+              Counter::kIs1024);
+  state.counters["request_throughput"] =
+      Counter(request_processed, Counter::kIsRate);
+  state.counters["request_latency"] =
+      Counter(request_processed, Counter::kIsRate | Counter::kInvert);
+  state.counters["message_throughput"] =
+      Counter(message_processed, Counter::kIsRate);
+  state.counters["message_latency"] =
+      Counter(message_processed, Counter::kIsRate | Counter::kInvert);
+
+  return absl::OkStatus();
+}
+
+// Helper function for benchmarking single bytes payload translation from JSON.
+void BM_SinglePayloadFromJson(::benchmark::State& state,
+                              uint64_t payload_length,
+                              bool streaming,
+                              uint64_t stream_size) {
+  std::string json_msg = absl::StrFormat(
+      R"({"payload" : "%s"})", GetRandomBytesString(payload_length, true));
+  absl::Status status = BenchmarkJsonTranslation(state,
+                                                 kBytesPayloadMessageType,
+                                                 json_msg,
+                                                 streaming,
+                                                 stream_size,
+                                                 1);
+  if (!status.ok()) {
+    state.SkipWithError(status.ToString().c_str());
+  }
+}
+
+static void BM_SinglePayloadFromJsonNonStreaming(::benchmark::State& state) {
+  BM_SinglePayloadFromJson(state, state.range(0), false, 0);
+}
+
+static void BM_SinglePayloadFromJsonStreaming(::benchmark::State& state) {
+  BM_SinglePayloadFromJson(state,
+                           kBytesPayloadLengthForStreaming,
+                           true,
+                           state.range(0));
+}
+
+// Helper function for benchmarking int32 array payload translation from JSON.
+void BM_Int32ArrayPayloadFromJson(::benchmark::State& state,
```

Comment:
can we not to prefix the functions with "BM_"  if they are not used directly inside the BENCHMARK macro,  it will be easier to read